src/components/TheCanvas.vue

```diff
@@ -2,293 +2,101 @@
+import CanvasSockets from '@/mixins/CanvasSockets'
+import StageWatcher from '@/mixins/StageWatcher'
 @Component({
-  name: 'TheCanvas'
+  name: 'TheCanvas',
+  mixins: [CanvasSockets, StageWatcher]
 })
 
-export default class TheCanvas extends Vue {
-  @Prop() private id!: string
-  @Tools.Action(ToolsAction.DISABLE) disable!: () => void
-  @Action(`tools/${ToolsAction.ENABLE}`) enable!: () => void
-  @Action(`tools/${ToolsAction.DISABLE_TOOL}`) disableTool!: () => void
-  @Action(`tools/${ToolsAction.ENABLE_TOOL}`) enableTool!: (name: string) => void
+export default class TheCanvas extends mixins(CanvasSockets, StageWatcher) {
+  @Prop() id!: string
   @Getter(`tools/${ToolGetters.ENABLED_TOOL}`) enabledTool!: Tool
-  @Getter(`tools/${ToolGetters.ENABLED}`) enabled!: boolean
   @Getter(`tools/${ToolGetters.TOOLS}`) tools!: Tool[]
-  // @Sockets.Getter(SocketGetters.SOCKET) socket!: WebSocket
-  // @Sockets.Action(SocketActions.SEND_IF_OPEN) send!: (message: string) => void
   @Action(`canvas/${CanvasAction.ADD_CANVAS_ELEMENT}`) addCanvasElement!: (canvasElement: CanvasElement) => void
-  @Action(`canvas/${CanvasAction.ADD_CANVAS_ELEMENT_HISTORY}`) addCanvasElementHistory!: (canvasElement: CanvasElement) => void
   @Action(`canvas/${CanvasAction.HIDE_CANVAS_ELEMENT}`) hideCanvasElement!: (payload: HideCanvasElementInterface) => void
-  @Action(`canvas/${CanvasAction.DELETE_CANVAS_ELEMENT}`) deleteCanvasElement!: (canvasElement: CanvasElement) => void
   @Getter(`canvas/${CanvasGetters.CANVAS_ELEMENTS}`) canvasElements!: CanvasElement[]
-  @Getter(`canvas/${CanvasGetters.CANVAS_ELEMENTS_HISTORY}`) canvasElementsHistory!: CanvasElement[]
   @Getter(`stage/${StageGetters.STAGE_ZOOM}`) stageZoom!: number
   @Getter(`stage/${StageGetters.STAGE_CONFIG}`) stageConfig!: CustomStageConfig
-  @Action(`stage/${StageActions.SET_CONFIG}`) setStageConfig!: (config: CustomStageConfig) => void
-  @Action(`stage/${StageActions.SET_DIMENSIONS}`) setDimensions!: (dimensions: { width: number; height: number }) => void
-  @Action(`stage/${StageActions.SCALE}`) setScale!: (scale: number) => void
-
-  canvasElement: CanvasElement = {
-    jti: 'SAM',
-    id: '',
-    data: [],
-    tool: {
-      name: ''
-    },
-    layerId: Math.random().toString(36),
-    tracker: Tracker.ADDITION,
-    change: false,
-    hasMoved: false,
-    position: { x: 0, y: 0 }
-  }
-
-  mouseDown = false;
+  @Action(`layer/${LayerActions.LAYER_SET}`) setLayer!: (layer: Konva.Layer) => void
+  @Getter(`layer/${LayerGetters.LAYER}`) layerNode!: Konva.Layer
+  @Action(`stage/${StageActions.SET_STAGE}`) setStage!: (stage: Konva.Stage) => void
+  @Getter(`stage/${StageGetters.STAGE}`) stage!: Konva.Stage
+  @Getter(`canvasEntity/${CanvasEntityGetters.CANVAS_ENTITY}`) canvasEntity!: CanvasEntityState
 
   $refs!: {
     layer: VueKonvaLayer;
     stage: VueKonvaStage;
   }
 
-  created () {
-    window.onload = () => {
-      this.setStageConfig({
-        scale: {
```

Comment:
all of this shit has been extracted to their own files to handle their own small but specific tasks